plot_sparsities.py

```diff
@@ -0,0 +1,238 @@
+import argparse
+import json
+import logging
+import os
+import time
+import math
+from collections import defaultdict, Counter
+from typing import Dict, List, Optional, Tuple, Set
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch.utils.data import DataLoader, Dataset
+from transformers import (
+    AutoTokenizer, 
+    AutoModelForCausalLM,
+    set_seed
+)
+from datasets import load_dataset
+import matplotlib.pyplot as plt
+from tqdm import tqdm
+import pickle
+
+from src.activation_capture import ACTIVATION_CAPTURE
+
+# Setup logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+class ContextualSparsityAnalyzer:
```

Comment:
Shall we move this file to utilities??

Reply:
we could, i just put it in the same place as measure_contextual_sparsities was because its basically the same thing. I didn't want to delete the old one though because I didn't fully understand what its purpose was.